source/crud/tests/unified/bulkWrite-delete-hint-serverError.yml

```diff
@@ -0,0 +1,145 @@
+# This file was created automatically using mongodb-spec-converter.
+# Please review the generated file, then remove this notice.
+
+description: bulkWrite-delete-hint-serverError
+schemaVersion: '1.1'
+createEntities:
+  -
+    client:
+      id: &client0 client0
+      useMultipleMongoses: false
+      uriOptions:
+        retryReads: false
+      observeEvents:
+        - commandStartedEvent
+  -
+    database:
+      id: &database0 database0
+      client: client0
+      databaseName: &database_name crud-v2
+  -
+    collection:
+      id: &collection0 collection0
+      database: database0
+      collectionName: &collection_name BulkWrite_delete_hint
+runOnRequirements:
+  -
+    minServerVersion: 3.4.0
+    maxServerVersion: 4.3.3
+initialData:
+  -
+    collectionName: *collection_name
+    databaseName: *database_name
+    documents:
+      -
+        _id: 1
+        x: 11
+      -
+        _id: 2
+        x: 22
+      -
+        _id: 3
+        x: 33
+      -
+        _id: 4
+        x: 44
+tests:
+  -
+    description: 'BulkWrite deleteOne with hints unsupported (server-side error)'
+    operations:
+      -
+        object: *collection0
+        name: bulkWrite
+        arguments:
+          requests:
+            -
+              deleteOne:
+                filter: &deleteOne_filter1
+                  _id: 1
+                hint: &hint_string _id_
+            -
+              deleteOne:
+                filter: &deleteOne_filter2
+                  _id: 2
+                hint: &hint_doc
+                  _id: 1
+          ordered: true
+        expectError:
+          isError: true
```

Comment:
This would be a good use of `isClientError: false`, but I realize it's outside the scope of this PR. Perhaps the best course of action would be a ticket to go through and revise the converted tests after the fact. I know we don't use `commandFailedEvent` very much in spec tests, but that may be another thing to explore for server-side error tests.

Reply:
Yes, I noticed that the error expectations in the unified format are significantly more advanced than what we had. It's difficult for the converter to make these changes and I didn't want to stray too far from the automated files. We might be able to use the presence of a `result` field to decide that we're expecting a server-side error, but can't really use the absence of one to expect a client-side error (as the test may be expecting a server-side error without checking the command result).

Reply:
Feel free to log a ticket for improving the error expectations and I can triage it down the line.

Reply:
I created [DRIVERS-1698](https://jira.mongodb.org/browse/DRIVERS-1698) to track the effort.